expanderr.go

```diff
@@ -387,16 +392,52 @@ func (e *expansion) typeCheck(pkgname string, files []*ast.File) error {
 			}
 		}
 	}
+	return nil
+}
 
-	e.ce = callExprAtPath(e.path)
-	if e.ce == nil {
-		return fmt.Errorf("no ast.CallExpr found")
+// this function either returns just the return values of the function or, if there are no errors
+// returned, will also add in the no-error-return-template
+func (e *expansion) getFinalOutput(noReturnTempl *template.Template, errName string) (ast.Stmt, error) {
+	var normalReturn ast.Stmt = &ast.ReturnStmt{Results: e.results}
+
+	var returnsError = true
+	if res := e.caller.Results; res == nil {
+		returnsError = false
+	} else if list := res.List; len(list) == 0 {
+		returnsError = false
+	} else if id, ok := list[len(list)-1].Type.(*ast.Ident); ok && id.Name != "error" {
+		returnsError = false
 	}
-	e.callee, err = signatureOf(e.info, e.ce)
-	return err
+
+	if returnsError {
+		return normalReturn, nil
+	}
+
+	if noReturnTempl == nil { // default output when no error retuned
+		return &ast.ExprStmt{X: &ast.Ident{Name: "panic(err)"}}, nil
```

Comment:
I only noticed this now: here and when evaluating the template, you are essentially stuffing Go code into an ast.Ident.

While this might work fine today, it seems brittle and incorrect. For the literal here, it’s easy to change. Can you please look into parsing the template output into ast nodes, too?

Reply:
I looked into parsing the template into ast. The `go/parser` package doesn't seem to support any multiline statements that are not part of an entire file. Basically if its fine to assume that anyone would ever only want a single to do what they want with the error, and that if you're not using the default panic you'd like a return statement as well, we can get rid of the template all together and just let the user pass in a single statement that we can parse and always put in the return after.